src/capture_RPiHQ.cpp

```diff
@@ -1721,46 +1773,104 @@ if (extraFileAge == 99999 && ImgExtraText[0] == '\0') ImgExtraText = "xxxxxx   k
 					{
-						bool result = cv::imwrite(fileName, pRgb, compression_params);
-						if (! result) printf("*** ERROR: Unable to write to '%s'\n", fileName);
+						bool result = cv::imwrite(full_filename, pRgb, compression_parameters);
+						if (! result) printf("*** ERROR: Unable to write to '%s'\n", full_filename);
 					}
 				}
 			}
 			else
 			{
-				printf(" >>> Unable to take picture, return code=%d\n", (retCode >> 8));
+				numErrors++;
+				int r = retCode >> 8;
+				printf(" >>> Unable to take picture, return code=%d, r=%d\n", retCode, r);
+#if 1 == 1
+r = retCode;
+if (WIFSIGNALED(r)) r = WTERMSIG(r);
+{
+
+#else
+				if (r > 128)
+				{
+					// Got a signal.  See if it's one we care about.
+					r -= 128;
+#endif
+					std::string z = "";
+					if (r == SIGINT) z = "SIGINT";
+					else if (r == SIGTERM) z = "SIGTERM";
+					else if (r == SIGUSR1) z = "SIGUSR1";
+					else if (r == SIGHUP) z = "SIGHUP";
+					if (z != "")
+					{
+						printf("xxxx Got %s in capture_RPiHQ.cpp\n", z.c_str());
+						closeUp(98);
+					}
+					else
+					{
+						printf("xxxx Got signal %d in capture_RPiHQ.cpp\n", r);
+					}
+				}
 				Log(1, "  > Sleeping from failed exposure: %.1f seconds\n", (float)currentDelay_ms / MS_IN_SEC);
 				usleep(currentDelay_ms * US_IN_MS);
 				continue;
 			}
 
-			// Check for night time
-			if (dayOrNight == "NIGHT")
-			{
-				// Preserve image during night time
-				system("scripts/saveImageNight.sh &");
-			}
-			else
-			{
-				// Upload and resize image when configured
-				system("scripts/saveImageDay.sh &");
-			}
+			char cmd[1100];
+			char tmp[50];
+			Log(1, "  > Saving %s image '%s'\n", darkframe ? "dark" : dayOrNight.c_str(), final_file_name);
+			snprintf(cmd, sizeof(cmd), "scripts/saveImage.sh %s '%s'", dayOrNight.c_str(), full_filename);
+			snprintf(tmp, sizeof(tmp), " EXPOSURE_US=%ld", last_exposure_us);
+			strcat(cmd, tmp);
+			snprintf(tmp, sizeof(tmp), " BRIGHTNESS=%d", currentBrightness);
+			strcat(cmd, tmp);
+			snprintf(tmp, sizeof(tmp), " MEAN=%.6f", mean);
+			strcat(cmd, tmp);
+			snprintf(tmp, sizeof(tmp), " AUTOEXPOSURE=%d", currentAutoExposure ? 1 : 0);
+			strcat(cmd, tmp);
+			snprintf(tmp, sizeof(tmp), " AUTOGAIN=%d", currentAutoGain ? 1 : 0);
+			strcat(cmd, tmp);
+			snprintf(tmp, sizeof(tmp), " AUTOWB=%d", AutoAWB ? 1 : 0);
+			strcat(cmd, tmp);
+			snprintf(tmp, sizeof(tmp), " WBR=%1.2f", WBR);
+			strcat(cmd, tmp);
+			snprintf(tmp, sizeof(tmp), " WBB=%1.2f", WBB);
+			strcat(cmd, tmp);
+
+			// TEMPERATURE, GAIN, and BIN are used by the dark* scripts to sort and compare against
+			// prior values, so make them fixed width to aid in doing that.
+
+			// There's currently no way to get to the RPiHQ camera's temperature sensor.
+			//snprintf(tmp, sizeof(tmp), " TEMPERATURE=%02d", (int)round(actualTemp/10));
+			//strcat(cmd, tmp);
+			snprintf(tmp, sizeof(tmp), " GAIN=%03d", (int)round(currentGain));
```

Comment:
let's use two variables GAIN (factor for RPiHQ) and GAINDB (0.1db for ZWO) Users can decide which value they want to use.

Reply:
Sorry, we have to use the last gain:

```
long last_exposure_us		= 0;		// last exposure taken
float last_gain		= 0.0;		// last gain taken
```
```
					last_exposure_us = currentExposure_us;
					last_gain = currentGain;	// to be compatible with ZWO - ZWO gain=0.1 dB , RPiHQ gain=factor
					if (myModeMeanSetting.mode_mean)
						last_gain =  myRaspistillSetting.analoggain;
``````

					if (showGain == 1)
					{
						sprintf(bufTemp, "Gain: %1.2f", last_gain);
```
```
			snprintf(tmp, sizeof(tmp), " GAIN=%1.2f", last_gain);
			strcat(cmd, tmp);
			snprintf(tmp, sizeof(tmp), " GAINDB=%03d", (int)round(20.0 * 10.0 * log10(last_gain))); // ZWO gain 0.1dB

```

Reply:
I implemented the last_gain changes above, and added GAINDB to capture_RPiHQ.cpp.  I left capture.cpp alone.